Example/Pellicola/PhotoStarViewController.m

```diff
@@ -68,17 +70,41 @@ - (void)updateMessage {
 }
 
 - (NSInteger)collectionView:(nonnull UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
-    return [_images count];
+    return [_urls count];
 }
 
 - (nonnull __kindof UICollectionViewCell *)collectionView:(nonnull UICollectionView *)collectionView cellForItemAtIndexPath:(nonnull NSIndexPath *)indexPath {
     ImageCell *cell = (ImageCell *)[collectionView dequeueReusableCellWithReuseIdentifier:@"ImageCell"
                                                                              forIndexPath:indexPath];
-    
-    cell.imageView.image = _images[indexPath.item];
+    CGSize size = cell.imageView.bounds.size;
+    NSURL *imageURL = _urls[indexPath.item];
+    __weak typeof(self) weakSelf = self;
+    dispatch_async(dispatch_get_global_queue( DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^(void){
+        UIImage *image = [UIImage imageWithData:[NSData dataWithContentsOfURL:imageURL]];
+        UIImage *resizedImage = [weakSelf imageWithImage:image scaledToFillSize:size];
+        dispatch_async(dispatch_get_main_queue(), ^(void){
```

Comment:
⚠️ You should cancel this operation in `prepareForReuse` otherwise you will end up setting the image on an already reused cell